src/main/java/com/code_intelligence/jazzer/mutation/mutator/lang/FloatingPointMutatorFactory.java

```diff
@@ -0,0 +1,127 @@
+import com.code_intelligence.jazzer.mutation.api.SerializingMutator;
+import com.code_intelligence.jazzer.mutation.mutator.libfuzzer.LibFuzzerMutator;
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.lang.reflect.AnnotatedType;
+import java.util.Optional;
+import java.util.function.Predicate;
+
+final class FloatingPointMutatorFactory extends MutatorFactory {
+  @Override
+  public Optional<SerializingMutator<?>> tryCreate(AnnotatedType type, MutatorFactory factory) {
+    if (!(type.getType() instanceof Class)) {
+      return Optional.empty();
+    }
+    Class<?> clazz = (Class<?>) type.getType();
+
+    if (clazz == float.class || clazz == Float.class) {
+      return Optional.of(new FloatMutator(type, -Float.MAX_VALUE, Float.MAX_VALUE));
+    } else if (clazz == double.class || clazz == Double.class) {
+      return Optional.of(new DoubleMutator(type, -Double.MAX_VALUE, Double.MAX_VALUE));
+    } else {
+      return Optional.empty();
+    }
+  }
+
+  private static final class FloatMutator extends SerializingMutator<Float> {
+    FloatMutator(AnnotatedType type, Float defaultMinValueForType, Float defaultMaxValueForType) {}
+
+    public Float mutateWithLibFuzzer(Float value) {
+      return LibFuzzerMutator.mutateDefault(value, this, 0);
+    }
+
+    @Override
+    public Float init(PseudoRandom prng) {
+      return prng.closedRange(-Float.MAX_VALUE, Float.MAX_VALUE);
+    }
+
+    @Override
+    public Float mutate(Float value, PseudoRandom prng) {
+      return prng.closedRange(-Float.MAX_VALUE, Float.MAX_VALUE);
+    }
+
+    @Override
+    public Float read(DataInputStream in) throws IOException {
+      return in.readFloat();
+    }
+
+    @Override
+    public void write(Float value, DataOutputStream out) throws IOException {
+      out.writeFloat((float) value);
+    }
+
+    @Override
+    public Float detach(Float value) {
+      return value;
+    }
+
+    @Override
+    public String toDebugString(Predicate<Debuggable> isInCycle) {
+      return "Float";
+    }
+  }
+
+  private static final class DoubleMutator extends SerializingMutator<Double> {
+    DoubleMutator(
+        AnnotatedType type, Double defaultMinValueForType, Double defaultMaxValueForType) {}
+
+    public Double mutateWithLibFuzzer(Double value) {
+      return LibFuzzerMutator.mutateDefault(value, this, 0);
+    }
+
+    @Override
+    public Double init(PseudoRandom prng) {
+      return prng.closedRange(-Double.MAX_VALUE, Double.MAX_VALUE);
+    }
+
+    @Override
+    public Double mutate(Double value, PseudoRandom prng) {
+      return prng.closedRange(-Double.MAX_VALUE, Double.MAX_VALUE);
+    }
+
+    @Override
+    public Double read(DataInputStream in) throws IOException {
+      return in.readDouble();
+    }
+
+    @Override
+    public void write(Double value, DataOutputStream out) throws IOException {
+      out.writeDouble((double) value);
```

Comment:
This should also compile:
```suggestion
      out.writeDouble(value);
```

Reply:
Done!